examples/citi_bike/online_lp/citi_bike_lp.py

```diff
@@ -0,0 +1,187 @@
+                )
+
+                for neighbor_idx in range(self._num_neighbor):
+                    self._transfer[tick][station][neighbor_idx] = LpVariable(
+                        name=f"T{tick}_Transfer_from{station}_to{neighbor_idx}th", lowBound=0, cat="Integer"
+                    )
+
+        # Initialize the current inventory.
+        for station in range(self._num_station):
+            self._inventory[0][station] = init_inventory[station]
+
+    def _add_constraints(self, problem: LpProblem, demand: np.ndarray, supply: np.ndarray):
+        for tick in range(self._num_tick):
+            for station in range(self._num_station):
+                problem += (
+                    self._fulfillment[tick][station] <= demand[tick, station]
+                ), f"Fulfillment_Limit_T{tick}_S{station}"
+                # For intermediate variables.
+                problem += (
+                    self._transfer_from[tick][station] == lpSum(
+                        self._transfer[tick][station][neighbor_idx] for neighbor_idx in range(self._num_neighbor)
+                    )
+                ), f"TotalTransferFrom_T{tick}_S{station}"
+                problem += (
+                    self._transfer_to[tick][station] == lpSum(
+                        self._transfer[tick][neighbor][self._station_neighbor_list[neighbor].index(station)]
+                        for neighbor in range(self._num_station)
+                        if station in self._station_neighbor_list[neighbor][:self._num_neighbor]
+                    )
+                ), f"TotalTransferTo_T{tick}_S{station}"
+
+        for tick in range(1, self._num_tick):
+            for station in range(self._num_station):
+                problem += (
+                    self._inventory[tick][station] == (
+                        self._inventory[tick - 1][station]
+                        + supply[tick - 1, station] - self._fulfillment[tick - 1][station]
+                        + self._transfer_to[tick - 1][station] - self._transfer_from[tick - 1][station]
+                    )
+                ), f"Inventory_T{tick}_S{station}"
+                problem += (
+                    self._safety_inventory[tick][station] <= (
+                        self._inventory[tick - 1][station]
+                        + supply[tick - 1, station] - self._fulfillment[tick - 1][station]
+                        - self._transfer_from[tick - 1][station]
+                    )
+                ), f"SafetyInventory_T{tick}_S{station}"
+
+    def _set_objective(self, problem: LpProblem):
+        fulfillment_gain = lpSum(
+            math.pow(self._fulfillment_time_decay_factor, tick) * lpSum(
+                self._fulfillment[tick][station] for station in range(self._num_station)
+            ) for tick in range(self._num_tick)
+        )
+
+        safety_inventory_reward = self._safety_inventory_reward_factor * lpSum(
```

Comment:
no decay for this objective?

Reply:
I can understand the decay for fulfillment will more like to fulfill the near demand, do we need to do a similar thing for the safety_nventory.

Reply:
not set now. I agree that adding a time decay for this reward is better.